### PR TITLE
Gives the Chef a Beacon to pick their ingredient box

### PR DESCRIPTION
**Pull request by XRandomXManX**
## About The Pull Request

Adds a new beacon, the Ingredient Beacon which allows players to choose their ingredient sets, as any proper chef should!

## Why It's Good For The Game

This will allow chefs to make more of what they want to make.
All boxes are equally likely to be picked, but even if there were 'rare' ones I think the fun of SS13 should be about being able to try new things and do what you want rather than keeping Lootboxes that give you the Italian box for the fifth time!.
The beacon is very simple and works like the Instrument Beacon, it has been playtested many times.
If you do like surprises, however, you can now pick the 'random' option, which will choose a box for you.

## Changelog
:cl: Radacitus
add: Adds a new Ingredient Beacon and gives it to the Chef in-place of their Ingredient Box on spawn.
add: Adds some custom text that appears when you use the Chef and Curator Beacons.
/:cl:

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by wesoda25** [comment]
I think a lot of the fun comes from its randomness; it’s liberating to let rng determine your gimmick. If you’re set on adding options, could you add a random one too?

**Comment by optimumtact** [comment]
this is the dumbest thing

**Comment by XRandomXManX** [comment]
> I think a lot of the fun comes from its randomness; it’s liberating to let rng determine your gimmick. If you’re set on adding options, could you add a random one too?

Wildcard chooses from seven random ingredients: chili, tomato, carrot, potato, sweet potato, apple, chocolate bar, cherry, banana, cabbage, soybeans, corn, plump helmet, chanterelle. That itself could be random enough, but if you want some real randomness just ask the Botanists for whatever they have. The box is just there to make your first two or three starting dishes while the Botanists start growing whatever they want, it won't impact your entire game (Unless it's a Wizard Round)

**Comment by LemonInTheDark** [comment]
It's somewhat like cloning right? The choices are there to force you to be creative, if you can chose to avoid that it ruins the point

**Comment by XRandomXManX** [comment]
> It's somewhat like cloning right? The choices are there to force you to be creative, if you can chose to avoid that it ruins the point

You can be creative after you've made the two or three dishes a box allows you to make before you have to rely on what the crew finds and the Botanists grow.

**Comment by Qustinnus** [comment]
I don't see the point of saying "it forces you to be creative" when the creativity is limited to a specific set of dishes. It just removes player agency.

**Comment by wesoda25** [comment]
Typically I follow the theme given to me through the box. If it’s carnivore, I try and focus primarily on meat for the rest of the round. Sweets, I stick with deserts. Even if you don’t adopt a cooking gimmick, it does restrict you to certain food types. Just picking wildcard does not help with a round long gimmick. 

I don’t think it’s asking much at all for you to add a random option. You would be maintaining the past randomness while also giving the option incase people want it. I personally think its hugbox to even add the options, but if you’re going to, keep the random aspect, even it it’s just a choice.

**Comment by brotherbeyondd** [comment]
As a chef player, I quite enjoy the random box I get given. It sometimes helps me pick a gimmick and whilst choosing it WOULD be fun, I prefer the current randomised box.

**Comment by XRandomXManX** [comment]
> As a chef player, I quite enjoy the random box I get given. It sometimes helps me pick a gimmick and whilst choosing it WOULD be fun, I prefer the current randomised box.

Guess I'll have to find out how to add a randomized option ;-;

**Comment by XRandomXManX** [comment]
every box is ordered alphabetically, even the 'random' option, don't @ me

**Comment by XRandomXManX** [comment]
nevermind that broke the entire thing, i didn't even put that close to the right spot, i must sleep

**Comment by XRandomXManX** [comment]
I just wanted to add a little bit of a cute touch since this very simple pr has devolved into Lootboxes: Mankind Divided ;-;

**Comment by LemonInTheDark** [comment]
I know you put a lot of effort into the pr, and I don't wanna drive you away over that, but I think adding a choice here detracts from the spice this feature is intended to provide, and I don't wanna take more of your time then I have to.

If no other maint is interested, I'll be closing this pr in a few days.

**Comment by Doosteroni** [comment]
What kinda "spice" are you on about brother? You high or something? He even added a random option dude

**Comment by XRandomXManX** [comment]
> I know you put a lot of effort into the pr, and I don't wanna drive you away over that, but I think adding a choice here detracts from the spice this feature is intended to provide, and I don't wanna take more of your time then I have to.
> 
> If no other maint is interested, I'll be closing this pr in a few days.

If you want spice you should be able to pick your own boxes instead of having random chance have the possibility of giving you something you already have or don't want or got last time.

The PR has 12 likes to 3 dislikes, clearly the community likes this PR and doesn't see it the same way you do.

**Comment by LemonInTheDark** [comment]
I think the upside to the way it works now is the forced randomness, yaknow? If you can choose what to pick you've gonna be more likely to go for the same few things over and over again.

I know ***surprise mechanics*** and all, but it feels like an intentional part of how the ingredients work.

Thoughts?

**Comment by Qustinnus** [comment]
forced randomness would be interesting if it changed the way you had to deal with cooking, but instead it just forces you to cook specific things, not necessarily interesting, especially when most of your ingredients for the round are in the hands of another job that might or might not care what you want.

**Comment by Acer202** [comment]
Honestly, I usually just throw out or ignore my box if it's not helpful to the idea I had in mind before the shift even started. (Fancy kitchen, barbecue joint, sandwich shop). If I get a grain box, that's usually just being tossed, either to the botanist or to the disposals chute. Getting the meats box is a nice bonus. I'd prefer choices rather than disappointment that I didn't get the box I wanted.

**Comment by cacogen** [comment]
didn't read but bikeshed

**Comment by XRandomXManX** [comment]
> I think the upside to the way it works now is the forced randomness, yaknow? If you can choose what to pick you've gonna be more likely to go for the same few things over and over again.
> 
> I know _**surprise mechanics**_ and all, but it feels like an intentional part of how the ingredients work.
> 
> Thoughts?

Chefs don't need to be forced to be more creative.
Optimizing the fun out of games is a common issue, but Chef isn't a job were somebody would even WANT to take the easy way out. People pick Chef because they want a roleplay job where they can actually do things, or want to make new or rare dishes, not so they can grind out the exact same dish over and over again, and even IF they wanted to keep remaking dishes, they'd just throw away their ingredients box and use what the botanist is making.

**Comment by XRandomXManX** [comment]
> didn't read but bikeshed

None of any of this post has been about bikeshedding.
Even people arguing about adding an optional random option had to do with functionality, be it somewhat insignificant functionality. Also please read before replying to stuff.

**Comment by LemonInTheDark** [comment]
Ahhhhh

I might be wrong about this, but I'll bite

File: code/game/objects/items/miscellaneous.dm [hunk 1]
```diff
@@ -78,6 +243,10 @@
 			hero_item_list[initial(A.name)] = A
 	return hero_item_list
 
+/obj/item/choice_beacon/hero/spawn_option(obj/choice,mob/living/M)
+	new choice(get_turf(M))
+	to_chat(M, "<span class='hear'>You hear something crackle from the beacon for a moment before a voice speaks. \"Please stand by for a message from Sophronia Broadcasting. Message as follows: <b>Please enjoy your Sophronia Broadcasting's 'History Comes Alive branded' Costume Set, exactly as shown in the hit show!</b> Message ends.\"</span>")
+
```
Comment (LemonInTheDark) [inline code/game/objects/items/miscellaneous.dm:249]: Please document all the changes you make in the pr, it's mostly so I can say gattcha if you hide a grenade in a pie somewhere or something.

File: code/game/objects/items/miscellaneous.dm [hunk 2]
```diff
@@ -64,6 +63,172 @@
...
+							  /obj/item/reagent_containers/food/snacks/grown/apple,
+							  /obj/item/reagent_containers/food/snacks/chocolatebar,
+							  /obj/item/reagent_containers/food/snacks/grown/cherries,
+							  /obj/item/reagent_containers/food/snacks/grown/banana,
+							  /obj/item/reagent_containers/food/snacks/grown/cabbage,
+							  /obj/item/reagent_containers/food/snacks/grown/soybeans,
+							  /obj/item/reagent_containers/food/snacks/grown/corn,
+							  /obj/item/reagent_containers/food/snacks/grown/mushroom/plumphelmet,
+							  /obj/item/reagent_containers/food/snacks/grown/mushroom/chanterelle)
+		new randomFood(src)
```
Comment (LemonInTheDark) [inline code/game/objects/items/miscellaneous.dm:113]: Define randomFood outside the for loop, ya only really need to create it once

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
A recommendation and a request for explanation, get em fixed up

File: code/game/objects/items/miscellaneous.dm (hunk 2 above)
Comment (XRandomXManX) [reply code/game/objects/items/miscellaneous.dm:113]: > Define randomFood outside the for loop, ya only really need to create it once

Oops, good catch. That was something left there by the original creator of the Wildcard Box. I'll fix that up.

**Comment by XRandomXManX** [comment]
Nevermind, everything broke. Probably worthy of another PR, since it didn't really have anything to do with my changes.

**Comment by LemonInTheDark** [comment]
Nah I just misread it, you're good. Tested in game, only thoughts are the beacon looks a bit grey, maybe add a new sprite for em?

**Comment by XRandomXManX** [comment]
> Nah I just misread it, you're good. Tested in game, only thoughts are the beacon looks a bit grey, maybe add a new sprite for em?

It's meant to be grey, I thought it fit the chef well, and the grey isn't used by any other beacons I know of.
If you have another color in mind I could change it, but I'll just keep it the same otherwise.

**Comment by LemonInTheDark** [comment]
Up to you, I'm not a sprite lead for very good reasons, all grey looks a bit wrong to me, but that's all I can say. If @Twaticus has no concerns, I think it's ready for merging.

**Comment by Twaticus** [comment]
chefs colors are pretty much just white/grey so i'd say its all good :+1:

**Pull request merged**